traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -608,43 +580,15 @@ func (ds *TODeliveryService) Read(h http.Header, useIMS bool) ([]interface{}, er
 		// NOTE: it's required to handle minor version cases in a descending >= manner
 		case version.Major > 3:
 			returnable = append(returnable, ds.RemoveLD1AndLD2())
-		case version.Major > 2 && version.Minor >= 1:
+		case version.Major == 3 && version.Minor >= 1:
 			returnable = append(returnable, ds.DowngradeToV31())
-		case version.Major > 2:
-			returnable = append(returnable, ds.DowngradeToV31().DeliveryServiceV30)
-		case version.Major > 1:
-			returnable = append(returnable, ds.DowngradeToV31().DeliveryServiceNullableV15)
 		default:
 			return nil, nil, fmt.Errorf("TODeliveryService.Read called with invalid API version: %d.%d", version.Major, version.Minor), http.StatusInternalServerError, nil
 		}
 	}
 	return returnable, nil, nil, errCode, maxTime
 }
 
-func UpdateV15(w http.ResponseWriter, r *http.Request) {
-	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, []string{"id"})
-	if userErr != nil || sysErr != nil {
-		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
-		return
-	}
-	defer inf.Close()
-
-	id := inf.IntParams["id"]
-
-	ds := tc.DeliveryServiceNullableV15{}
-	if err := json.NewDecoder(r.Body).Decode(&ds); err != nil {
-		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("malformed JSON: "+err.Error()), nil)
-		return
-	}
-	ds.ID = &id
-
-	res, status, userErr, sysErr := updateV15(w, r, inf, &ds)
```

Comment:
I believe you can also remove the `updateV15` function as well

Reply:
Removed `updateV15` in 55cb74b94c